packages/flutter/lib/src/widgets/navigation_toolbar.dart

```diff
@@ -83,15 +88,22 @@ const double _kMiddleMargin = 16.0;
 class _ToolbarLayout extends MultiChildLayoutDelegate {
   _ToolbarLayout({
     this.centerMiddle,
+    double requestedMiddleMargin,
     @required this.textDirection,
-  }) : assert(textDirection != null);
+  }) : assert(textDirection != null),
+       middleMargin = requestedMiddleMargin ?? _kMiddleMargin;
```

Comment:
why default it here instead of the other places?

Reply:
This goes all the way to AppBar. If there's no default there, it gets passed to NavigationToolbar as null overriding the static default. Defaulting it in the place it is actually used made more sense.

We could provide a default in AppBar but I still need to default it here as well in case this is used from other widgets.

Reply:
We usually try to provide defaults because when we eventually get non-nullable types, it'll probably be more efficient to have non-nullable arguments than nullable ones.

What you can do here though is have a static const on NavigationToolbar that gives the default, and then have all the places that have this argument specify that as their default. That way it's both self-documenting and non-null.

Reply:
Fair enough. Done!